scripts/sct_fmri_moco.py

```diff
@@ -313,27 +344,40 @@ def fmri_moco(param):
     param_moco.path_out = ''
     param_moco.todo = 'estimate_and_apply'
     param_moco.mat_moco = 'mat_groups'
-    moco.moco(param_moco)
-
-    # create final mat folder
-    sct.create_folder(mat_final)
-
-    # Copy registration matrices
-    sct.printv('\nCopy transformations...', param.verbose)
-    for iGroup in range(nb_groups):
-        for data in range(len(group_indexes[iGroup])):
-            sct.copy(os.path.join('mat_groups', 'mat.T' + str(iGroup) + ext_mat), mat_final + 'mat.T' + str(group_indexes[iGroup][data]) + ext_mat, verbose=param.verbose)
-
-    # Apply moco on all fmri data
-    sct.printv('\n-------------------------------------------------------------------------------', param.verbose)
-    sct.printv('  Apply moco', param.verbose)
-    sct.printv('-------------------------------------------------------------------------------', param.verbose)
-    param_moco.file_data = 'fmri'
-    param_moco.file_target = file_data + '_mean_' + str(0)
-    param_moco.path_out = ''
-    param_moco.mat_moco = mat_final
-    param_moco.todo = 'apply'
-    moco.moco(param_moco)
+    file_mat = moco.moco(param_moco)
+
+    # TODO: if g=1, no need to run the block below (already applied)
+    if param.group_size == 1:
+        # if flag g=1, it means that all images have already been corrected, so we just need to rename the file
+        sct.mv('fmri_averaged_groups_moco.nii', 'fmri_moco.nii')
+    else:
+        # create final mat folder
+        sct.create_folder(mat_final)
+
+        # Copy registration matrices
+        sct.printv('\nCopy transformations...', param.verbose)
+        for iGroup in range(nb_groups):
+            for data in range(len(group_indexes[iGroup])):
```

Comment:
enumerate

Reply:
fixed in ef52dd32eb9b816f95e392767a9695d0d2c0cc02